experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -365,17 +378,19 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 	/**
 	 * Spawn a new branch that is based off of the current state of this branch.
 	 * Changes made to the new branch will not be applied to this branch until the new branch is merged back in.
+	 * @param repairDataStoreProvider - a RepairDataStoreProvider that reflects the state of the new branch
 	 * @param anchors - an optional set of anchors that the new branch is responsible for rebasing
 	 */
 	public fork(
-		repairDataStoreProvider?: IRepairDataStoreProvider,
+		repairDataStoreProvider: IRepairDataStoreProvider,
 		anchors?: AnchorSet,
 	): SharedTreeBranch<TEditor, TChange> {
 		this.assertNotDisposed();
 		const fork = new SharedTreeBranch(
 			this.head,
 			this.changeFamily,
-			this.undoRedoManager?.clone(repairDataStoreProvider),
+			repairDataStoreProvider,
```

Comment:
Do we want to clone here?

Reply:
Nope, a clone keeps a reference to the same forest. We're expecting that the one passed in corresponds to the new forest for this branch.

Reply:
Sorry, I meant that cloning clones the forest as well so we would lose the reference to the new forest that gets stored on the fork.